src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -463,6 +468,75 @@ def test_poll_interval(self):
         self.assertEqual(str(ctx.exception), "--poll-interval must be a positive integer")
 
 
+class ClientEndpoint(SyncTestCase):
+    """
+    Tests related to the client-api-endpoint global config option
+    """
+
+    def setUp(self):
+        super(ClientEndpoint, self).setUp()
+        self.basedir = FilePath(self.mktemp())
+        self.nodedir = self.useFixture(
+            NodeDirectory(FilePath(self.mktemp()))
+        )
+
+    def test_convert_tcp(self):
+        """
+        a tcp: endpoint can be autoconverted
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:127.0.0.1:5555")
```

Comment:
I wonder what `u"tcp:555555"` does?  I expect it makes it into the database?  Not sure there's much to do about this here, just noting it.

Reply:
It does make it in, yeah :/ (I guess Twisted doesn't check port ranges etc?) .. but even if we e.g. tried to listen there (to see if it's a valid port on this OS), failing to do so isn't *necessarily* bad (e.g. maybe the port is just busy right now, or maybe it's completely invalid or maybe you haven't fixed AppArmor yet or ...). There will be errors that just won't be found/apparent until you `magic-folder run` anyway.

All that said though, probably the HTTP API for "change the config" *should* actually try and listen first and generate an error if it can't .. because if it can't then you'll be unable to change the config back to something that does work.